src/osd/PrimaryLogPG.cc

```diff
@@ -3801,8 +3801,9 @@ int PrimaryLogPG::do_extent_cmp(OpContext *ctx, OSDOp& osd_op)
     return result;
   }
 
-  if (read_op.outdata.length() != osd_op.indata.length())
-    return -EINVAL;
+  if (read_op.outdata.length() != osd_op.indata.length()) {
+    read_op.outdata.append_zero(osd_op.indata.length() - read_op.outdata.length());
```

Comment:
@wangzhengyong While that is true that RBD treats missing data as zeros, that might not always be the case. In the generic sense, it might be better to return the offset where the first mismatch occurs (be it caused by either data or truncation).